ios/CodePush/CodePush.m

```diff
@@ -455,9 +455,12 @@ - (void)rollbackPackage
 {
     NSError *error;
     NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];
-
-    // Write the current package's metadata to the "failed list"
-    [self saveFailedUpdate:failedPackage];
+    if (error) {
```

Comment:
That example just says to me that an error can occur without the error pointer being set. The example there also shows a `success` flag being returned by the method. I completely agree with that example there that the success flag should be checked.

However, you are saying that an error pointer can be set without an error actually happening, i fail to see why especially considering the method we are calling into is also implemented by us.